pkg/agents/team7/agent1/agent.go

```diff
@@ -79,99 +93,539 @@ func New(baseAgent *infra.Base) (infra.Agent, error) {
+	r2 := rand.Intn(11) - 5
+
+	a.behaviour.kindness += int(float64(a.personality.neuroticism*r1) / 50)
+	a.behaviour.greediness += int(float64(a.personality.neuroticism*r2) / 50)
+
+	if a.behaviour.kindness < 0 {
+		a.behaviour.kindness = 0
+	} else if a.behaviour.kindness > 100 {
+		a.behaviour.kindness = 100
+	}
+
+	if a.behaviour.greediness < 0 {
+		a.behaviour.greediness = 0
+	} else if a.behaviour.greediness > 100 {
+		a.behaviour.greediness = 100
+	}
+
+}
+
+func (a *CustomAgent7) manageFloor() {
+	if len(a.opMem.orderPrevFloors) == 0 || a.opMem.orderPrevFloors[len(a.opMem.orderPrevFloors)-1] != a.Floor() { //If day 1 or floor change
+
+		a.Log("checking floor change........................................................................................")
+		if a.Age() == 0 {
+			a.Log("Personality Makeup:", infra.Fields{"A": a.personality.agreeableness, "C": a.personality.conscientiousness, "E": a.personality.extraversion, "N": a.personality.neuroticism, "O": a.personality.openness})
+		}
+
+		// Reset Variables on Floor Change
+		a.opMem.currentDayonFloor = 1 // reset currentDay counter
+		a.opMem.receivedReq = false   // reset Requests
+		a.opMem.takeFood = 0
+		a.opMem.leaveFood = 0
 
+		// If the floor tracker is not empty
+		if len(a.opMem.orderPrevFloors) != 0 {
+			// ------------------ Run() Block A.1: Estimates the food available on the new floor based on past experience ------------------
+			a.foodNewFloor()
+
+		}
+		a.opMem.orderPrevFloors = append(a.opMem.orderPrevFloors, a.Floor()) //append current floor to floor tracker
+
+	} else { //increment currentDayonFloor counter
+		a.opMem.currentDayonFloor++
+	}
+}
+
+func (a *CustomAgent7) foodNewFloor() {
+	closestFloorAboveCurrent := 0
+	closestFloorBelowCurrent := math.Inf(1)
+	beenOnCurrentFloorBefore := false
+
+	for i := 0; i < len(a.opMem.orderPrevFloors); i++ {
+		if a.opMem.orderPrevFloors[i] == a.Floor() {
+			beenOnCurrentFloorBefore = true
+			break
+		}
+	}
+
+	if !beenOnCurrentFloorBefore {
+		for i := 0; i < len(a.opMem.orderPrevFloors); i++ {
+			if a.opMem.orderPrevFloors[i] < a.Floor() && a.opMem.orderPrevFloors[i] > closestFloorAboveCurrent {
+				closestFloorAboveCurrent = a.opMem.orderPrevFloors[i]
+			}
+		}
+		for i := 0; i < len(a.opMem.orderPrevFloors); i++ {
+			if a.opMem.orderPrevFloors[i] > a.Floor() && a.opMem.orderPrevFloors[i] < int(closestFloorBelowCurrent) {
+				closestFloorBelowCurrent = float64(a.opMem.orderPrevFloors[i])
+			}
+		}
+	}
+
+	var expectedFood int
+
+	if beenOnCurrentFloorBefore {
+		expectedFood = a.opMem.prevFloors[a.Floor()].avgFood
+	} else {
+
+		if closestFloorAboveCurrent != 0 && closestFloorBelowCurrent != math.Inf(1) {
+			expectedFood = a.opMem.prevFloors[int(closestFloorBelowCurrent)].avgFood + (a.opMem.prevFloors[closestFloorAboveCurrent].avgFood-a.opMem.prevFloors[int(closestFloorBelowCurrent)].avgFood)*(int(closestFloorBelowCurrent)-a.Floor())/(int(closestFloorBelowCurrent)-int(closestFloorAboveCurrent))
```

Comment:
This line is so long, can you split it into variables and/or write it across multiple lines?